libs/external-db-mysql/src/mysql_schema_provider.ts

```diff
@@ -47,35 +50,59 @@ export default class SchemaProvider implements ISchemaProvider {
                          [...(columns || []).map((c: { name: any }) => c.name)])
-                  .catch( translateErrorCodes )
+                  .catch( err => translateErrorCodes(err, collectionName) )
     }
 
     async drop(collectionName: string): Promise<void> {
         await this.query(`DROP TABLE IF EXISTS ${escapeTable(collectionName)}`)
-                  .catch( translateErrorCodes )
+                  .catch( err => translateErrorCodes(err, collectionName) )
     }
 
     async addColumn(collectionName: string, column: InputField): Promise<void> {
         await validateSystemFields(column.name)
         await this.query(`ALTER TABLE ${escapeTable(collectionName)} ADD ${escapeId(column.name)} ${this.sqlSchemaTranslator.dbTypeFor(column)}`)
-                  .catch( translateErrorCodes )
+                  .catch( err => translateErrorCodes(err, collectionName) )
+    }
+
+    async changeColumnType(collectionName: string, column: InputField): Promise<void> {
+        await validateSystemFields(column.name)
+        await this.query(`ALTER TABLE ${escapeTable(collectionName)} MODIFY ${escapeId(column.name)} ${this.sqlSchemaTranslator.dbTypeFor(column)}`)
+                  .catch( err => translateErrorCodes(err, collectionName) )
     }
 
     async removeColumn(collectionName: string, columnName: string): Promise<void> {
         await validateSystemFields(columnName)
         return await this.query(`ALTER TABLE ${escapeTable(collectionName)} DROP COLUMN ${escapeId(columnName)}`)
-                         .catch( translateErrorCodes )
+                         .catch( err => translateErrorCodes(err, collectionName) )
     }
 
-    async describeCollection(collectionName: string): Promise<ResponseField[]> {
-        const res = await this.query(`DESCRIBE ${escapeTable(collectionName)}`)
-                              .catch( translateErrorCodes )
-        return res.map((r: { Field: string; Type: string }) => ({ field: r.Field, type: r.Type }))
-                  .map( this.translateDbTypes.bind(this) )
+    async describeCollection(collectionName: string): Promise<Table> {
+        interface describeTableResponse {
+            Field: string,
+            Type: string,
+        }
+        
+        const res: describeTableResponse[] = await this.query(`DESCRIBE ${escapeTable(collectionName)}`)
+                                                       .catch( err => translateErrorCodes(err, collectionName) )
+        const fields = res.map(r => ({ field: r.Field, type: r.Type })).map(this.appendAdditionalRowDetails.bind(this))
+        return {
+            id: collectionName,
+            fields: fields as ResponseField[],
+            capabilities: this.collectionCapabilities(res.map(f => f.Field))
+        }
     }
 
-    translateDbTypes(row: ResponseField): ResponseField {
+    private appendAdditionalRowDetails(row: ResponseField) {
```

Comment:
This function doesn't get as a parameter `RespondField`, it returns it. @MXPOL

Reply:
We be fixed in #396